src/scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -199,6 +200,8 @@ def clone(
             else:
                 repo = clone_from()
             cls._set_default_tracking_branch(repo)
+        except HTTPUnauthorized:
+            raise AuthError(url)
```

Comment:
```suggestion
        except HTTPUnauthorized as exc:
            raise AuthError(url) from exc
```

Reply:
👍 Also changed other instances of AuthError